### PR TITLE
Fixes #2321 : I'm eating this should be in view mode, not edit mode

### LINKED ISSUE #2321: "I'm eating this" should be removed -> replaced by "Add to list"
image
I expected the button to be on the summary tab in view mode, not edit mode.

### PR DESCRIPTION
**Pull request by balaji-ramavathu**
## Description

Made "I'm eating this" checkbox in editing page from editing mode to view mode
Added a button in summary tab that toggles addition and removal of the product to eaten list.

## Related issues and discussion
#fixes #2321
 
 ## Screen-shots, if any
 
eating1
eating2

### CODE REVIEW

**Comment by teolemon** [comment]
@balaji-ramavathu can you make the "compare product" button the same size ?

**Comment by Karljoones** [comment]
Some small conflicts here, too

**Comment by teolemon** [comment]
@Karljoones Conflicts seem to have been fixed

**Comment by deniger** [comment]
A small UI issue:
image

**Comment by teolemon** [comment]
actually, we've completely revamped the Summary tab, so this should be brought in line with the new design.

**Comment by deniger** [comment]
We have 2 button in fact in the view:
image

We should keep the bottom one ?

File: app/src/main/res/layout/fragment_summary_product.xml [hunk 1]
```diff
@@ -136,6 +136,39 @@
...
+                android:layout_margin="8dp"
+                android:background="@drawable/rounded_button"
+                android:drawableLeft="@drawable/ic_check_white_24dp"
+                android:padding="8dp"
+                android:text="@string/txt_im_eating_this"
+                android:textColor="@color/white"
+                android:visibility="gone" />
+
+            <Button
+                android:drawableLeft="@drawable/ic_compare_arrows_black_18dp"
```
Comment (deniger) [inline app/src/main/res/layout/fragment_summary_product.xml]: There is an issue as the button with the id "@+id/compare_product_button" is defined twice in this file

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/views/AddProductActivity.java [hunk 2]
```diff
@@ -371,10 +371,6 @@ public void onError(Throwable e) {
                         saveProductOffline();
                     }
                 });
-        if(productDetails.get("eating").equals("true")){
```
Comment (deniger) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/views/AddProductActivity.java]: The line 1066... should be removed to and then the method "addProductToList" should be removed as not used

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/views/product/summary/SummaryProductFragment.java [hunk 3]
```diff
@@ -1033,6 +1048,76 @@ public void onCompareProductButtonClick() {
...
+        StringBuilder stringBuilder = new StringBuilder();
+        if (isNotEmpty(product.getBrands())) {
+            stringBuilder.append(product.getBrands());
+        }
+        if (isNotEmpty(product.getQuantity())) {
+            stringBuilder.append(" - ").append(product.getQuantity());
+        }
+        String productDetailsString = stringBuilder.toString();
+        String imageUrl = product.getImageUrl();
+        if (isNotEmpty(product.getImageUrl())) {
```
Comment (deniger) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/views/product/summary/SummaryProductFragment.java]: Lines 1090 -1092 should be removed as not usefull ( the stringBuilder is no more used, productDetailsString is already built and the image URL is set below)

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/views/product/summary/SummaryProductFragment.java [hunk 4]
```diff
@@ -1033,6 +1048,76 @@ public void onCompareProductButtonClick() {
...
+    }
+
+    @OnClick(R.id.btn_eating_ticked)
+    public void onBtnEatingTickedClick() {
+        btnEating.setVisibility(View.VISIBLE);
+        btnEatingTicked.setVisibility(View.GONE);
+        removeProductFromEatenlist();
+    }
+
+    public boolean checkIfProductIsEaten() {
```
Comment (deniger) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/views/product/summary/SummaryProductFragment.java]: could be renamed to isProductEatenByCurrentUser

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/views/product/summary/SummaryProductFragment.java [hunk 5]
```diff
@@ -368,6 +373,16 @@ public void refreshView(State state) {
             mUrlImage = product.getImageUrl();
         }
 
+        yourListedProductsDao = Utils.getAppDaoSession(getContext()).getYourListedProductDao();
+        //check if this product exists in eaten list
+        if (checkIfProductIsEaten()) {
+            btnEating.setVisibility(View.GONE);
```
Comment (deniger) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/views/product/summary/SummaryProductFragment.java]: please, is it possible to add a new method
updateEatingButtons(boolean isEaten){
...
} 
and replace the lines 378-384 by updateEatingButton(isProductEatenByCurrentUser())
and then 1053-1054 updateEatingButtons(true)
and then 1060-1061 updateEatingButtons(false)

**Comment by deniger** [comment]
@balaji-ramavathu please, could you have a look to this ?

**Comment by balaji-ramavathu** [comment]
@teolemon  sorry to say this, but right now I'm so busy working on my GSoC proposal to other organisations and also on my academic stuff. So I don't think I have time to work on this. I hope you understand. I loved my previous experience of working with you for this project, so I ensure you that I will come back and contribute in the future :-)

@deniger thanks for the review, but I'm closing this PR for the above mentioned reason. You may proceed to fix this issue if you want and you can take my code from here and make changes to it rather than starting from the beginning :-)

**Pull request closed without merging**